test/req.get.js

```diff
@@ -12,6 +16,12 @@ describe('req', function(){
         assert(req.get('Something-Else') === undefined);
         res.end(req.get('Content-Type'));
       });
+      
+      app.use(function(req, res){
+        var name = ['something'];
+        assert.throws(typeThrowError(name), TypeError, "Error thrown");
```

Comment:
This test is just testing the function `typeThrowError` in this file, and is not actually testing the new `if` statement you added in `lib/request.js`. This is the reason your pull request is in a failure state, because you haven't actually tested your change yet :) To test your change, you'd want to do something like `req.header(42)` and test that it threw.